packages/editor/src/lib/editor/managers/HistoryManager.ts

```diff
@@ -117,9 +115,9 @@ export class HistoryManager<
 			this._batchDepth++
 			if (this._batchDepth === 1) {
 				transact(() => {
-					const mostRecentActionId = this._undos.get().head?.id
+					const mostRecentAction = this._undos.get().head
 					fn()
-					if (mostRecentActionId !== this._undos.get().head?.id) {
+					if (mostRecentAction !== this._undos.get().head) {
```

Comment:
This is the only place where we were using the id... but we can do a regular comparison instead.